src/System.ObjectModel/src/System/Collections/ObjectModel/ObservableCollection.cs

```diff
@@ -123,9 +126,64 @@ protected override void RemoveItem(int index)
 
             OnCountPropertyChanged();
             OnIndexerPropertyChanged();
-            OnCollectionChanged(NotifyCollectionChangedAction.Remove, removedItem, index);
+
+            if (!_skipCollectionChanged)
+            {
+                OnCollectionChanged(NotifyCollectionChangedAction.Remove, removedItem, index);
+            }
         }
 
+        /// <summary>
+        /// Called by base class Collection&lt;T&gt; when a count of items is removed from the list;
+        /// raises a CollectionChanged event to any listeners.
+        /// </summary>
+        protected override void RemoveItemsRange(int index, int count)
+        {
+            CheckReentrancy();
+
+            int length = this.Count;
+            if (index < this.Count - count)
+            {
+                length = index + count;
+            }
+
+            T[] removedItems = new T[length];
+            for (int i = index; i < length; i++)
+            {
+                removedItems[i] = this[index];
+            }
+
+            _skipCollectionChanged = true;
+            try
+            {
+                base.RemoveItemsRange(index, count);
+            }
+            finally
+            {
+                _skipCollectionChanged = false;
+            }
+
+            if (count > 0)
+            {
+                OnCountPropertyChanged();
+                OnIndexerPropertyChanged();
+                OnCollectionChanged(NotifyCollectionChangedAction.Remove, removedItems, index);
```

Comment:
I don't think this call is correct. It is calling the following overload:

https://github.com/dotnet/corefx/blob/e45615101e28fac1a89414e701fee4e4be84a4fc/src/System.ObjectModel/src/System/Collections/ObjectModel/ObservableCollection.cs#L263-L266

which calls:

https://github.com/dotnet/corefx/blob/e45615101e28fac1a89414e701fee4e4be84a4fc/src/System.ObjectModel/src/System/Collections/Specialized/NotifyCollectionChangedEventArgs.cs#L72

which is intended for when a single item changes.

In this case, multiple items have changed. `removedItems` is incorrectly being passed as the single item that changed.

Instead, I believe it should be creating a `NotifyCollectionChangedEventArgs` using the following constructor:

https://github.com/dotnet/corefx/blob/e45615101e28fac1a89414e701fee4e4be84a4fc/src/System.ObjectModel/src/System/Collections/Specialized/NotifyCollectionChangedEventArgs.cs#L138

which allows passing the changed items as a collection.

Reply:
I updated the `OnCollectionChanged` method to handle `IList` correctly. Let me know if there is anything else for this.